### DISCUSSION TITLE
How to assign (update) a value as a list (array) or a dict

### DISCUSSION THREAD

**Discussion by juanpons** [mikefarah/yq#1236] [Q&A] [answered]
I am using **yq** to update helm chart yaml files. 
Sometimes I require that the value assigned is a list or a dict. 
For instance, I may want to change from:
```
ingress:
  hosts: [ ]
```
to:
```
ingress:
  hosts:
    - host: chart.domain.com
      paths: [ ] 
```

I could do that easily in version 3, however I have not yet found how to do it in version 4.

Ideally this may work:
```
yq -i   ' .ingress.hosts = "[ { host: chart.domain.com, paths: [ ] }  ]"    ingress.yaml
```
However this sets the ingress hosts variable to a text string, instead of setting it to a list of dicts.

If I remove the outer " " , I get error that the [ is unexpected.

Any ideas? 

Help kindly appreciated!

**Comment by mikefarah** [comment, marked as answer]
What you need to do is append to the array. Putting everything in double quotes will set it as a string.

I'm not at a machine with `yq` installed, but it should look something like:

```
yq -i   ' .ingress.hosts += {"host": chart.domain.com, "paths": []}'
```

Note that quotes are required around the string key name (as you can use an expression for the key, just as you do for the value).

**Reply by juanpons** [reply]
Thanks Mike! 
It worked with a minor change (add commas to chart.domain,com) 
```
yq -i   ' .ingress.hosts += {"host": "chart.domain.com", "paths": []}'
```
KR, Juan
